### DISCUSSION TITLE
Designated Initializers and Older Compilers

### DISCUSSION THREAD

**Discussion by matthewdcong** [wjakob/nanobind#108] [Q&A] [answered]
First of all, thank you for nanobind! I've been working with it (along with pybind11) for the past month or so and I've had a great experience so far.

nanobind fails to compile on Visual Studio 2017 and GCC 7.3.1 (the compiler version in the Red Hat Developer Toolset for RHEL 7) because it uses C++20 designated initializers for the `PyType_Spec` structs in `nb_type.cpp` and `nb_internals.cpp`. Replacing these with aggregate initializers allows nanobind to compile on these older compilers.

What's the general feeling towards submitting a PR in this vein? On one hand, I understand that these compilers are relatively old now (and long due for an upgrade) but on the other hand this is a localized change that will facilitate the adoption of nanobind, especially since pytorch, etc. work with these older compilers.

**Comment by wjakob** [comment, marked as answer]
I would accept such a change, it seems quite minor. Whenyou switch over to aggregate initializers, can you add comments indicating the field names? E.g.

```cpp
static Foo value {
    .bar = ..
};
```

to
```cpp
static Foo value {
    /* bar = */ ...
};
```

**Reply by matthewdcong** [reply]
Sure thing, thanks @wjakob!

**Reply by matthewdcong** [reply]
@wjakob: Addressed in https://github.com/wjakob/nanobind/pull/115

**Reply by wjakob** [reply]
(Merged in the meantime)
